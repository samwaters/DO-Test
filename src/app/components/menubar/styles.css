.menubar {
    align-items: center;
    border-bottom: 1px solid var(--secondary);
    display: flex;
    height: 64px;
    padding: 0 32px 0 16px;
}

.menubar-left {
    align-items: center;
    display: flex;
    flex: 1;
    gap: 10px;
}

.menubar-logo {
    font-size: 28px;
}
.menubar-title {
    font-size: 22px;
}

.menubar-right {
    display: flex;
    flex-direction: row;
    gap: 10px;
    justify-content: flex-end;
    width: 250px;
}