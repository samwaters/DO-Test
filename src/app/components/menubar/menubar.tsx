import "./menubar.styles.css"
import { TiFlowMerge } from "react-icons/ti"
import { IconButton } from "@/app/components/ui/iconbutton"
import { MdFileDownload, MdOutlineSave } from "react-icons/md"
import { useDownload } from "@/app/hooks/download"
export const MenuBar = () => {
    const download = useDownload()

    return (
        <nav className="menubar">
            <div className="menubar-left">
                <div className="menubar-logo">
                    <TiFlowMerge />
                </div>
                <div className="menubar-title">Untitled Workflow</div>
            </div>
            <div className="menubar-right">
                <IconButton icon={<MdOutlineSave />} id="btn-save" size="medium" tooltip="Save Changes" />
                <IconButton
                    icon={<MdFileDownload />}
                    id="btn-download"
                    onClick={download}
                    size="medium"
                    tooltip="Download as PNG"
                />
            </div>
        </nav>
    )
}
