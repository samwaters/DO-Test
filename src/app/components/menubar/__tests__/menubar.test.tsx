import { render, screen } from "@testing-library/react"
import { MenuBar } from "@/app/components/menubar/menubar"
import { ReactFlowProvider } from "@xyflow/react"

describe("MenuBar", () => {
    it("Renders the menu bar", () => {
        render(
            <ReactFlowProvider>
                <div data-testid="menubar">
                    <MenuBar />
                </div>
            </ReactFlowProvider>,
        )
        expect(screen.getByTestId("menubar")).toMatchSnapshot()
    })
})
