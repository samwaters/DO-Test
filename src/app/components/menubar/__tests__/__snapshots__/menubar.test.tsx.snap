// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar Renders the menu bar 1`] = `
<div
  data-testid="menubar"
>
  <nav
    class="menubar"
  >
    <div
      class="menubar-left"
    >
      <div
        class="menubar-logo"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 16.184v-1.851c0-1.93-1.57-3.5-3.5-3.5-.827 0-1.5-.673-1.5-1.5v-1.517c1.161-.415 2-1.514 2-2.816 0-1.654-1.346-3-3-3s-3 1.346-3 3c0 1.302.839 2.401 2 2.815v1.518c0 .827-.673 1.5-1.5 1.5-1.93 0-3.5 1.57-3.5 3.5v1.851c-1.161.415-2 1.514-2 2.816 0 1.654 1.346 3 3 3s3-1.346 3-3c0-1.302-.839-2.401-2-2.816v-1.851c0-.827.673-1.5 1.5-1.5.979 0 1.864-.407 2.5-1.058.636.651 1.521 1.058 2.5 1.058.827 0 1.5.673 1.5 1.5v1.851c-1.161.415-2 1.514-2 2.816 0 1.654 1.346 3 3 3s3-1.346 3-3c0-1.302-.839-2.401-2-2.816zm-11 3.816c-.552 0-1-.449-1-1s.448-1 1-1 1 .449 1 1-.448 1-1 1zm5-16c.552 0 1 .449 1 1s-.448 1-1 1-1-.449-1-1 .448-1 1-1zm5 16c-.552 0-1-.449-1-1s.448-1 1-1 1 .449 1 1-.448 1-1 1z"
          />
        </svg>
      </div>
      <div
        class="menubar-title"
      >
        Untitled Workflow
      </div>
    </div>
    <div
      class="menubar-right"
    >
      <button
        class="icon-button medium"
        data-tooltip-content="Save Changes"
        data-tooltip-id="btn-save"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M17 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm2 16H5V5h11.17L19 7.83V19zm-7-7c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zM6 6h9v4H6z"
          />
        </svg>
      </button>
      <button
        class="icon-button medium"
        data-tooltip-content="Download as PNG"
        data-tooltip-id="btn-download"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
        </svg>
      </button>
    </div>
  </nav>
</div>
`;
